userspace/libsinsp/CMakeLists.txt

```diff
@@ -311,3 +316,43 @@ if (BUILD_LIBSINSP_EXAMPLES)
 	add_subdirectory(examples)
 endif()
 
+# Build our pkg-config "Libs:" flags. For now, loop over SINSP_LIBRARIES. If
+# we ever start using pkg_search_module or pkg_check_modules in cmake/modules
+# we could add each module to our "Requires:" line instead. We might need to
+# expand this to use some of the techniques in
+# https://github.com/curl/curl/blob/curl-7_84_0/CMakeLists.txt#L1539
+set(SINSP_PKG_CONFIG_LIBS)
+set(SINSP_PKG_CONFIG_LIBDIRS)
+foreach(sinsp_lib ${SINSP_LIBRARIES})
+	if(${sinsp_lib} MATCHES "^-")
+		# We have a flag. Pass it through unchanged.
+		list(APPEND SINSP_PKG_CONFIG_LIBS ${sinsp_lib})
+	elseif(${sinsp_lib} MATCHES "/")
+		# We have a path. Convert it to -L<dir> + -l<lib>.
+		get_filename_component(sinsp_lib_dir ${sinsp_lib} DIRECTORY)
+		list(APPEND SINSP_PKG_CONFIG_LIBDIRS -L${sinsp_lib_dir})
+		get_filename_component(sinsp_lib_base ${sinsp_lib} NAME_WE)
+		string(REGEX REPLACE "^lib" "" sinsp_lib_base ${sinsp_lib_base})
+		list(APPEND SINSP_PKG_CONFIG_LIBS -l${sinsp_lib_base})
+	elseif(${sinsp_lib} STREQUAL "scap")
+		# We require libscap.pc, so skip it.
+	else()
+		# Assume we have a plain library name. Prefix it with "-l".
+		list(APPEND SINSP_PKG_CONFIG_LIBS -l${sinsp_lib})
+	endif()
+endforeach()
+
+# Build our pkg-config "Cflags:" flags.
+set(SINSP_PKG_CONFIG_INCLUDES)
+foreach(sinsp_include_directory ${LIBSINSP_INCLUDE_DIRS})
+	if(NOT sinsp_include_directory MATCHES "(^${PROJECT_SOURCE_DIR}|^${PROJECT_BINARY_DIR})")
+		list(APPEND SINSP_PKG_CONFIG_INCLUDES -I${sinsp_include_directory})
+	endif()
+endforeach()
+
+string(REPLACE ";" " " SINSP_PKG_CONFIG_LIBS "${SINSP_PKG_CONFIG_LIBS}")
+list(REMOVE_DUPLICATES SINSP_PKG_CONFIG_LIBDIRS)
```

Comment:
```
CMake Error at userspace/libsinsp/CMakeLists.txt:356 (list):
  list sub-command REMOVE_DUPLICATES requires list to be present.
```

@geraldcombs I guess `SINSP_PKG_CONFIG_LIBDIRS` is not a list if `elseif(${sinsp_lib} MATCHES "/")` branch is never entered.

Reply:
Weird. I can't duplicate this with the following minimal test case

```
cmake_minimum_required(VERSION 2.8.5)
project(TEST)
set(L)
list(REMOVE_DUPLICATES L)
```

I initialized SINSP_PKG_CONFIG_LIBDIRS and SINSP_PKG_CONFIG_INCLUDES with an empty string. Maybe that will fix it?